### ISSUE TITLE
Cannot install/reauthenticate Alexa Integration - DO NOT USE 4.13.8! Solution: Revert to 4.13.7

### ISSUE THREAD

**Issue by oywino** [alandtse/alexa_media_player#2675]
**IMPORTANT: Please search the issues, including closed issues, and the [FAQ](https://github.com/alandtse/alexa_media_player/wiki/FAQ) before opening a new issue. The template is mandatory; failure to use it will result in issue closure.**

**Describe the bug**

After upgrading HA fram 2024.10.3 to 2024.11.1 I got a log error message saying that the Alexa Integration could not be loaded (see below). So I tried to re-install the integration which eventually required also deleting and re-creating the 2FA with  a new key.
Now, when completing the installation, I get a success message which tells me to ignore the "Aborted message" from HA. But there is no "Aborted message" anywhere!
When I click "close" on the success message, there's no integration added, and no new messages at all in the log.
When I restart HA, I get the below error message in the log all over again.

**To Reproduce**

**Expected behavior**

**Screenshots**

**System details**

- Home Assistant version: 2024.11.1
- alexa_media version (from `const.py` or HA startup log): 4.13.8
- alexapy version (from `pip show alexapy` in homeasssistant container or HA startup log):
- Is Amazon [2FA/2SV enabled](https://github.com/alandtse/alexa_media_player/wiki/Configuration#enable-two-factor-authentication-for-your-amazon-account)  (y/n): y
- Amazon Domain: amazon.com

**Debug Logs (alexa_media & alexapy)**
Please provide [logs](https://github.com/alandtse/alexa_media_player/wiki/FAQ#how-do-i-enable-debug-logging-for-the-component).

**Additional context**

Logger: homeassistant.setup
Source: setup.py:416
First occurred: 8:20:11 PM (1 occurrences)
Last logged: 8:20:11 PM

Error during setup of component alexa_media
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/setup.py", line 416, in _async_setup_component
    result = await task
             ^^^^^^^^^^
  File "/config/custom_components/alexa_media/__init__.py", line 159, in async_setup
    CONF_PUBLIC_URL: account[CONF_PUBLIC_URL],
                     ~~~~~~~^^^^^^^^^^^^^^^^^
KeyError: 'public_url'

**Comment by sihartley** [comment]
Exact Same Issue here after HA upgrade to 2024.11.1
Config will not accept Public URL, even if hardcoded into __init__.py line 190

**Comment by duquedaniel** [comment]
Same error here, after upgrading alexa integration.

Logger: homeassistant.setup
Source: setup.py:416
First occurred: 08:59:38 (1 occurrences)
Last logged: 08:59:38

Error during setup of component alexa_media
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/setup.py", line 416, in _async_setup_component
    result = await task
             ^^^^^^^^^^
  File "/config/custom_components/alexa_media/__init__.py", line 159, in async_setup
    CONF_PUBLIC_URL: account[CONF_PUBLIC_URL],
                     ~~~~~~~^^^^^^^^^^^^^^^^^
KeyError: 'public_url'

**Comment by eabraga69** [comment]
Mesmo problema por aqui. Integração Amazon alexa 4.13.8 não carrega após atualização do HA para a nova versão 2024.11.1 .

**Comment by VolkerLaude** [comment]
Exactly the same problem here:

  File "/config/custom_components/alexa_media/__init__.py", line 159, in async_setup
    CONF_PUBLIC_URL: account[CONF_PUBLIC_URL],
                     ~~~~~~~^^^^^^^^^^^^^^^^^

**Comment by thearchitectuk** [comment]
I am also getting the same now :-( Not sure how to remediate it either.

![image](https://github.com/user-attachments/assets/0659bf8c-a318-4f8c-9b99-9acd1afa5aa0)

**Comment by mesut3000** [comment]
Same error here :

Logger: homeassistant.setup
Source: setup.py:416
First occurred: 12:19:43 (1 occurrences)
Last logged: 12:19:43

Error during setup of component alexa_media
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/setup.py", line 416, in _async_setup_component
    result = await task
             ^^^^^^^^^^
  File "/config/custom_components/alexa_media/__init__.py", line 159, in async_setup
    CONF_PUBLIC_URL: account[CONF_PUBLIC_URL],
                     ~~~~~~~^^^^^^^^^^^^^^^^^
KeyError: 'public_url'

**Comment by danielbrunt57** [comment]
Do not install 4.13.8!
Revert to AMP 4.13.7 and wait for 4.13.9 to come out.
If HA renotifies you about 4.13.8, select skip to make it go away.

**Comment by thearchitectuk** [comment]
Does anyone know how I go about downgrading this integration? It's not something I've ever done before! Thanks

**Comment by danielbrunt57** [comment]
HACS (HA sidebar) > search "alexa" > Alexa Media Player > Overflow menu (3 vertical dots) > Redownload > Need a different version? > expand Release > Select v4.13.7
After download, restart HA to effect change.

**Comment by heftved** [comment]
Same fix as above, moved back to 4.13.7 Ironically the 4.13.8 update says it fixes the "Missing Public URL" issue! Clearly it did not!

**Comment by danielbrunt57** [comment]
> Same fix as above, moved back to 4.13.7 Ironically the 4.13.8 update says it fixes the "Missing Public URL" issue! Clearly it did not!

No, it made things worse.  It is being reverted to 4.13.7 while I examine the whole mess surrounding public_url which actually dates back to when it became a configurable option (Aug/Sept?).  Legacy YAML was never considered then as it was planned to be deprecated.  4.13.8 tried to fix things for YAML for its last few days but UI config became messed up.  The plan now is announce it will be deprecated as of 4.14.0 and with no fixes for it between now and then.
